### PR TITLE
When using CMake, always also build single-precision variant.

### LINKED ISSUE #41: Please consider suggesting an "official" suffix for single-precision builds
Currently, double-precision and single-precision builds default to sharing the same library name.  As a consequence, downstream packagers (linux, conda) typically just package the double-precision version, and the single precision version is unavailable.  In fact, it is not that difficult to provide both single and double precisions in e.g. a linux package; the single-precision library can simply use a separate name, and this is readily achieved with e.g. `cmake ... -DBUILD_SHARED_LIBS=ON -DCMAKE_C_FLAGS=-D__cminpack_float__ -DCMAKE_RELEASE_POSTFIX=s` which builds the single precision shared lib as `libcminpacks.so` (the "s" suffix being the one suggested in https://www.math.utah.edu/software/minpack.html#linking-local too).
Would you be willing to declare "officially" that it is preferred that downstream packagers provide both single and double precision versions of the library, with the single-precision using the "s" suffix (for example)?

### PR DESCRIPTION
**Pull request by anntzer**
Closes #41 (per https://github.com/devernay/cminpack/issues/41#issuecomment-622253014).
(Only the second commit matters for the feature, but cleaning the style helped me understand the logic :))

### CODE REVIEW

**Review by devernay — CHANGES_REQUESTED** [review]
Thank you!

I'm OK with this change, but can you move the doc about building with cmake, and the fact that it builds both libraries, is a separate section of the doc as indicated below?

I would also like feedback from other contributors (@xantares @bherisse  @tadeu @chaplin89): Is everyone OK with systematically building the 2 versions?

Also, why not the "long double" (128-bit float) version? The "half" version is really just a proof-of-concept, but "long double" may be useful if precision is an issue. long double is not supported by every compiler/platform so there would have to be a test for that. I'm not a CMake afficionado, so I don't know the proper way of testing this, but the FFTW configure script does this test.

**Comment by devernay** [comment]
Also, it's OK to reformat the CMakelist, but it's easier to review if it's a separate PR.

**Comment by xantares** [comment]
Are you sure you need both variants at the same time ? What's the use-case ?
Maybe a cmake option to select the variant is what we need.
I dont want to build the lib twice.

**Comment by anntzer** [comment]
> Are you sure you need both variants at the same time ? What's the use-case ?

The use case is when cminpack is packaged by redistributors, which include at least linux distros and conda.  In that case they will typically just provide a "plain" build of the library, which here means only the double-precision version.  I would rather have them provide all precisions together, so that I can easily have whatever precision I need (single, in my case) on machines with no compiler available.

> Maybe a cmake option to select the variant is what we need.  I dont want to build the lib twice.

Per the above I believe the default should be to provide all of them (because otherwise I'd need to go to each and every redistributor and convince them that they should do a non-default build of the library), but we could reasonably add a CMake flag to just build one of them.

File: docs/index.html [hunk 1]
```diff
@@ -117,7 +117,7 @@ <h3>Download</h3>
 <h3>Using CMinpack</h3>
 
 <p>The CMinpack calls have the same name as the FORTRAN functions, in lowercase (e.g. <code>lmder(...)</code>). See the links to the documentation below, or take a look at the simple examples in the <code>examples</code> directory of the distribution. The simple examples are named after the function they call: <code>tlmder.c</code> is the simple example for <code>lmder</code>.</p>
-<p>If you want to use the single precision CMinpack, you should define __cminpack_float__ before including <code>cminpack.h</code>. __cminpack_half__ has to be defined for the half-precision version (and the code needs to be compiled with a C++ compiler).</p>
+<p>If you want to use the single precision CMinpack, you should define __cminpack_float__ before including <code>cminpack.h</code>. __cminpack_half__ has to be defined for the half-precision version (and the code needs to be compiled with a C++ compiler). By default, building via CMake will generate both a double precision librarty, named <code>cminpack</code>, and a single precision library, named <code>cminpacks</code>.</p>
```
[the remark this answers is not in the document]
Comment (anntzer) [reply docs/index.html]: done

**Comment by devernay** [comment]
@xantares are you satisfied by @anntzer 's replies?

To me it looks OK:
- only one build
- two libraries are installed (could be three, including the long double version which IMHO would be a valuable addition, but we can leave that for later)

**Comment by devernay** [comment]
can you rebase and push the changes?

**Review by devernay — APPROVED** [review]

**Comment by anntzer** [comment]
done

**Comment by devernay** [comment]
Side note: @xantares and @anntzer would you agree to be collaborators on this project? This would allow you to merge PRs, but they would still need to be reviewed and approved by someone else.

**Comment by devernay** [comment]
@anntzer I'm curious, do you have an automatic formatter for CMakeLists or did you do it by hand?

**Comment by anntzer** [comment]
I did it by hand plus some regexes...

**Comment by anntzer** [comment]
Thanks for the offer of being a collaborator.  I don't claim any expertise in the relevant field (I really just needed a reasonable implementation of `lmdif`) and already have much on my plate, but can try to occasionally have a look if you need help with something.  No time guarantees, though :-)

**Comment by anntzer** [comment]
@devernay Any chance for a new release with these changes?  Thanks!

**Comment by devernay** [comment]
Done!

**Comment by anntzer** [comment]
Thanks!

**Pull request merged**